packages/frontend-plugin-api/api-report.md

```diff
@@ -320,29 +320,14 @@ export { BackstageIdentityApi };
 
 export { BackstageIdentityResponse };
 
-// @public (undocumented)
-export interface BackstagePlugin<
+// @public @deprecated (undocumented)
+export type BackstagePlugin<
   TRoutes extends AnyRoutes = AnyRoutes,
   TExternalRoutes extends AnyExternalRoutes = AnyExternalRoutes,
   TExtensionMap extends {
     [id in string]: ExtensionDefinition;
   } = {},
-> {
-  // (undocumented)
-  readonly $$type: '@backstage/BackstagePlugin';
-  // (undocumented)
-  readonly externalRoutes: TExternalRoutes;
-  // (undocumented)
-  getExtension<TId extends keyof TExtensionMap>(id: TId): TExtensionMap[TId];
-  // (undocumented)
-  readonly id: string;
-  // (undocumented)
-  readonly routes: TRoutes;
-  // (undocumented)
-  withOverrides(options: {
-    extensions: Array<ExtensionDefinition>;
-  }): BackstagePlugin<TRoutes, TExternalRoutes, TExtensionMap>;
-}
+> = FrontendPlugin<TRoutes, TExternalRoutes, TExtensionMap>;
```

Comment:
Would it have been more stringent to make this `FrontendPlugin<...> & { $$type: '@backstage/BackstagePlugin' }`? Might as well, right

Reply:
<img width="251" alt="image" src="https://github.com/user-attachments/assets/1ffc15ab-216d-49da-8a02-18df129730f2">

Gets tricky, but I think we want to accept both versions though in `createApp` or there'll be a bunch of type errors when importing old plugins

Reply:
That's kind of where I'm at too - will we have actual type errors if we try to "mask" the difference here. Sure, maybe an Omit is needed instead of my handwavy TS verbatim :)